Port NASSP to OpenOrbiter

CaptainSwag101 opened the pull request:
For the future of NASSP's development, I think it would be wise to start planning to build for OpenOrbiter, since it has several bugfixes and other improvements that may make development easier. Here are what needs to be done to make this happen:

- [x] Fix building with OpenOrbiter's SDK
- [x] Switch from OrbiterSound to XRSound
- [x] Fix remaining XRSound bugs (specific sounds like non-RCS engines were originally handled directly by OrbiterSound, and their functionality will either need to be moved to XRSound or done directly in NASSP now.)
- [ ] Fix certain D3D9Client issues, such as the EMS scroll texture causing crashes in the 3D VC, and the Floodlights not working
- [x] Fix non-looping sounds like switches etc. not working if a user loads into a scenario starting with the 3D VC, right now those sounds don't work until switching to the 2D cockpit at least once per scenario load.
- [x] Fix the 32-bit moons of Mars, Uranus, and Neptune causing corruption and destroying vessel orbits (remove them from our configs)
- [ ] Any other bugs found after the time of writing.

ThymoNL commented:
I have seen patches that add CMake support to NASSP. Those will also be a big help to making NASSP's development future proof.

CaptainSwag101 committed:
Merge branch 'XRSound' of https://github.com/indy91/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Project build release configuration overhaul

CaptainSwag101 committed:
Merge branch 'XRSound' of https://github.com/CaptainSwag101/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Rip out VESIM and all DirectSound/DirectInput code

CaptainSwag101 committed:
Fix MFD pointer obliteration

CaptainSwag101 committed:
Merge from upstream

CaptainSwag101 committed:
More project build directory tweaks
This should hopefully mean that all projects use VS/MSBuild's helpful variables for determining intermediate directory and project output name based on project configuration details, such as debug vs release, project name, etc.

CaptainSwag101 committed:
Change MFD process handle to OAPI_MSGTYPE

CaptainSwag101 committed:
Change build toolkit to VS2019 instead of VS2022

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Use the new lunar gravity model, bump coefficient cutoff to 120 from default 10

CaptainSwag101 committed:
More detailed comment about lunar gravity coefficient change

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Switch to L1 lunar gravity model for RTCC

CaptainSwag101 committed:
Bump Earth gravity model to 50 coefficient cutoff

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'RTCCFix' of https://github.com/indy91/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'XRSound' of https://github.com/indy91/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'XRSound' of https://github.com/CaptainSwag101/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter3

CaptainSwag101 committed:
Fix building Debug profile for OO Debug x64

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'XRSound' of https://github.com/CaptainSwag101/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'OpenOrbiter' of https://github.com/CaptainSwag101/NASSP into OpenOrbiter

CaptainSwag101 committed:
Disable Mars/Uranus/Neptune moons to fix trajectory chaos on x64

CaptainSwag101 committed:
Add build dependencies of PanelSDK for s1b and Skylab projects

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge XRSound volume changes

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/CaptainSwag101/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge from upstream, failures rework

CaptainSwag101 committed:
Merge vibration modifier update

rcflyinghokie committed:
Add R2 Padloads

rcflyinghokie committed:
Correct 14 padloads to use Artemis values

CaptainSwag101 committed:
Merged SPS line heater/flange temp changes

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge pull request #1 from rcflyinghokie/R2Padloads

Add R2 Padloads

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge latest changes

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Update to IMU Drift changes

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

CaptainSwag101 committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into OpenOrbiter

maxq519 commented:
Has this PR been replaced by https://github.com/orbiternassp/NASSP/pull/1288, or is this still being maintained?

CaptainSwag101 commented:
This is no longer being maintained. Thanks for the reminder, I'll close this PR.
